zilliqa/src/contracts/deposit.sol

```diff
@@ -421,6 +435,13 @@ contract Deposit {
         }
     }
 
+    // Returns the next block number at which new stakers are added,
+    // existing ones removed and/or deposits of existing stakers change
+    function nextUpdate() public view returns (uint256 blockNumber) {
+        if (latestComputedEpoch > currentEpoch())
+            blockNumber = latestComputedEpoch * blocksPerEpoch;
```

Comment:
Why is this not `blockNumber = (currentEpoch() + 2) * blocksPerEpoch`? That's the logic we use in `deposit` and `unstake` so shouldn't we be consistent?

Reply:
Not really imo. If there are future changes then `latestComputedEpoch` determines when they become effective. This is either at the beginning of `currentEpoch() + 2` or if this function is called later, maybe already at the beginning of `currentEpoch() + 1`.

Reply:
Note that the function is `public`, i.e. it's not only intended for internal use when we emit the events.

Reply:
Oh yes, I misunderstood the point of this function. Fair enough :)